src/main/java/com/aws/greengrass/clientdevices/auth/api/UseCases.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.api;
+
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.dependency.Context;
+
+import javax.inject.Inject;
+
+
+public class UseCases {
+    private final Context context;
+
+    public interface UseCase<R, D> {
+        @SuppressWarnings("PMD.SignatureDeclareThrowsException")
+        R execute(D dto) throws Exception;
+    }
```

Comment:
We have `CrashableFunction` in the Nucleus which is the same thing (more or less)

Reply:
Just saw it. The only issue with that interface is that forces the Exception generic type to be defined and if we want to use it for a function that does not throw any exception it wouldn't work (please correct me if I am wrong). For example

```
class Foo implements CrashableFuntion<Void, Void, Exception> {
     @Override
     public Void apply(Void a) {
         System.out.println("Hello);
    }
}
```

would complain saying we need to throw an exception or pass something as the 3rd generic param

Reply:
You should be able to just use a lambda `(Void a) -> { sout; return null }`.